### DISCUSSION TITLE
[Dataset edit modal] UX for edit metrics tab need improvement

### DISCUSSION THREAD

**Discussion by rosemarie-chiu** [apache/superset#19193] [Ideas]
**Is your feature request related to a problem? Please describe.**
It is confusing how to edit metrics in edit dataset modal.
1. I can only add SQL expression in the list, it isn't a field on the expanded form?
![CleanShot 2021-07-14 at 17 16 29](https://user-images.githubusercontent.com/52086618/125708820-3c6d773b-beea-490b-b662-7c08f1182dc8.gif)

2. I can only add label after I expand it? It is unclear to user that they need to expand the item to add label vs can update sql expression right from the list
![CleanShot 2021-07-14 at 17 20 23](https://user-images.githubusercontent.com/52086618/125709043-abf6ffc9-8369-4610-8e6d-93aad5f794fc.gif)

3. Once added a label, can edit it on either list or expanded form, but they are not synced?
![CleanShot 2021-07-14 at 17 21 45](https://user-images.githubusercontent.com/52086618/125709130-bf2d86f0-3212-487e-af89-19368d622c51.gif)

**Describe the solution you'd like**
1. I think if we want to expand an item into form and have more input fields, we should be consistent that sql expression field, label field all present in the expand form.
2. Should be consistent on either able to edit every fields in list view
3. Should not be able to update data in list when the form is expanded (problem 3)

**Describe alternatives you've considered**
Alternatively, make it not able to edit any fields in list view at all, do all modification in the expand form.

**Comment by junlincc** [comment]
let's discuss a middle ground solution and tackle it by mid Aug. @michael-s-molina 🙏

**Comment by michael-s-molina** [comment]
@rosemarie-chiu Totally agree with your points. I would like to add that we should consider Metrics, Columns, and Calculated Columns when deciding the behavior because they all look the same.

**Comment by srinify** [comment]
Converting to Discussion since it's a feature request!
